### PR TITLE
Add function run_ldb_tool

### PR DESCRIPTION
**Pull request by solotzg**
Integrate [`Ldb Tool`](https://github.com/facebook/rocksdb/wiki/Administration-and-Data-Access-Tool) [`ldb_tool.cc`](https://github.com/facebook/rocksdb/blob/master/tools/ldb_tool.cc) of RocksDB.
`run_ldb_tool` encapsulates `LDBTool::Run`, which can accept the argument from cmd line.

### CODE REVIEW

File: tests/test_tool.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+use tempdir::TempDir;
+
+#[inline]
+fn generate_vec_string(s: &String) -> Vec<String> {
+    let d: Vec<_> = s
+        .split(' ')
+        .collect::<Vec<_>>()
+        .iter()
+        .map(|s| s.to_string())
+        .collect();
```
Comment (kennytm) [inline tests/test_tool.rs]: The first `collect::<Vec<_>>()` is redundant, just use 

```rust
s.split(' ').map(|s| s.to_string()).collect()
```

Comment (solotzg) [reply tests/test_tool.rs]: done.

File: librocksdb_sys/crocksdb/c.cc [hunk 2]
```diff
@@ -4680,4 +4685,36 @@ uint64_t crocksdb_perf_context_env_new_logger_nanos(crocksdb_perf_context_t* ctx
   return ctx->rep.env_new_logger_nanos;
 }
 
+void crocksdb_run_ldb_tool(int argc, char** argv, char** errptr) {
+  errptr = nullptr;
```
Comment (huachaohuang) [inline librocksdb_sys/crocksdb/c.cc]: So why pass the `errptr` in?

Comment (solotzg) [reply librocksdb_sys/crocksdb/c.cc]: Just keep the code style equal with the way other functions use `ffi_try!`.

**Comment by solotzg** [comment]
@huachaohuang 
Initially, I wanted to put the related test cases in the `test` module, which made me export the unnecessary function.
Now, I have implemented all those functions inside the test module of `rocksdb.rs`

File: librocksdb_sys/crocksdb/c.cc (hunk 2 above)
Comment (huachaohuang) [reply librocksdb_sys/crocksdb/c.cc]: I think it is not necessary, why do we need to keep it consistent to use `ffi_try!`?

**Comment by solotzg** [comment]
@huachaohuang `LDBTool::Run` invokes `LDBCommandRunner::RunCommand` but don't return anything. I extracted the code from `LDBCommandRunner::RunCommand` as `crocksdb_run_ldb_cmd` for testing the cmd line.

File: librocksdb_sys/crocksdb/c.cc (hunk 2 above)
Comment (solotzg) [reply librocksdb_sys/crocksdb/c.cc]: I've fixed it.

**Pull request merged**